services/QuillLMS/spec/services/demo/report_demo_creator_spec.rb

```diff
@@ -4,14 +4,54 @@
 
 RSpec.describe Demo::ReportDemoCreator do
   let!(:teacher) {create(:teacher)}
+  let(:session_data) { Demo::SessionData.new }
+
+  # essentially using these as fixtures to test the demo data
+  let(:activity_id) { 1663 }
```

Comment:
Currently, `demo_config` is only coincidentally connected to the structure of `ACTIVITY_PACK_TEMPLATES`
Why not make the coupling to the actual constants explicit?  That way if the structure in `ACTIVITY_PACK_TEMPLATES` changes, the test will fail.  

```
let(:activity_pack_template) { ACTIVITY_PACKS_TEMPLATES.first }
let(:activity_pack_name) { activity_pack_template[:name] }
let(:activity_id) { activity_pack_template[:activity_ids].first }
let(:user_id) { activity_pack_template[:activity_sessions].first[activity_id] }

  let(:demo_config) do
    [
      {
        name:  activity_pack_template_name,
        activity_ids: [activity_id],
        activity_sessions: [
          {activity_id => user_id},
          {activity_id => user_id},
          {activity_id => user_id},
          {activity_id => user_id},
          {activity_id => user_id}
        ]
      }
    ]
  end
```

Reply:
That's a good point. I've updated the tests to have one that runs `create_demo` with the live config. And I've added some tests around the specific confiq structure to catch changes (note, this uncovered the fact that one of the 3 keys was redundant, so I removed it).

I feel good about the coverage now.